### PR TITLE
Spark on Mesos

### PR DESCRIPTION
**Pull request by yuchaoran2011**
A squash and merge of commits made in [fdp-hive-on-spark](https://github.com/lightbend/fdp-hive-on-spark) repo. Made it a fork of Hive. Updated to use the Spark service created using [fdp-spark-for-hive](https://github.com/lightbend/fdp-spark-for-hive-docker).

### CODE REVIEW

File: packaging/src/main/assembly/bin.xml [hunk 1]
```diff
@@ -434,3 +434,395 @@
 
 </assembly>
 
+=======
```
Comment (deanwampler) [inline packaging/src/main/assembly/bin.xml]: Is this "======" a git merge artifact?

File: pom.xml [hunk 2]
```diff
@@ -197,7 +197,7 @@
     <tez.version>0.9.1-SNAPSHOT</tez.version>
     <slider.version>0.92.0-incubating</slider.version>
     <super-csv.version>2.2.0</super-csv.version>
-    <spark.version>2.2.0</spark.version>
+    <spark.version>2.2.1</spark.version>
```
Comment (deanwampler) [inline pom.xml]: 2.2.1? The latest is 2.2.0, right?

File: ql/src/test/org/apache/hadoop/hive/ql/exec/TestExecDriver.java [hunk 3]
```diff
@@ -496,15 +490,13 @@ public void testMapPlan1() throws Exception {
     LOG.info("Beginning testMapPlan1");
     populateMapPlan1(db.getTable(Warehouse.DEFAULT_DATABASE_NAME, "src"));
     executePlan();
-    fileDiff("lt100.txt.deflate", "mapplan1.out");
```
Comment (deanwampler) [inline ql/src/test/org/apache/hadoop/hive/ql/exec/TestExecDriver.java]: Why did these disappear. It appears they do some sort of test validation.

File: packaging/src/main/assembly/bin.xml (hunk 1 above)
Comment (yuchaoran2011) [reply packaging/src/main/assembly/bin.xml]: Yeah good catch! Thanks!

File: pom.xml (hunk 2 above)
Comment (yuchaoran2011) [reply pom.xml]: 2.2.1 is available now but we are using 2.2.0 with hive. I'll change it to 2.2.0.

File: ql/src/test/org/apache/hadoop/hive/ql/exec/TestExecDriver.java (hunk 3 above)
Comment (yuchaoran2011) [reply ql/src/test/org/apache/hadoop/hive/ql/exec/TestExecDriver.java]: This is a squashed merge from @blublinsky change. I'll let him comment: https://github.com/lightbend/fdp-hive-on-spark/blob/master/ql/src/test/org/apache/hadoop/hive/ql/exec/TestExecDriver.java#L500

**Pull request merged**